File: archivist/src/main/java/com/zorroa/archivist/web/api/DyHierarchyController.java
package com.zorroa.archivist.web.api;

import com.zorroa.archivist.HttpUtils;
import com.zorroa.archivist.domain.DyHierarchy;
import com.zorroa.archivist.domain.DyHierarchySpec;
import com.zorroa.archivist.domain.Folder;
import com.zorroa.archivist.service.DyHierarchyService;
import com.zorroa.archivist.service.FolderService;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

/**
 * Created by chambers on 8/10/16.
 */
@RestController
public class DyHierarchyController {

    private static final Logger logger = LoggerFactory.getLogger(DyHierarchyController.class);

    @Autowired
    FolderService folderService;

    @Autowired
    DyHierarchyService dyHierarchyService;

    @RequestMapping(value="/api/v1/dyhi/_folder/{id}", method= RequestMethod.GET)
    public DyHierarchy getByFolder(@PathVariable int id) {
        Folder f = folderService.get(id);
        return dyHierarchyService.get(f);
    }

    @RequestMapping(value="/api/v1/dyhi", method= RequestMethod.POST)
    public DyHierarchy create(@RequestBody DyHierarchySpec spec) {
        return dyHierarchyService.create(spec);
    }

    @RequestMapping(value="/api/v1/dyhi/{id}", method= RequestMethod.POST)
    public Map<String, Object> delete(@PathVariable int id) {
        DyHierarchy dh = dyHierarchyService.get(id);
        boolean result = dyHierarchyService.delete(dh);
        return HttpUtils.status("DyHierarchy", id, "delete", result);
    }

    @RequestMapping(value="/api/v1/dyhi/{id}", method= RequestMethod.GET)
    public DyHierarchy get(@PathVariable int id) {
        return dyHierarchyService.get(id);
    }
}

File: archivist/src/main/kotlin/com/zorroa/archivist/web/api/DyHierarchyController.kt
package com.zorroa.archivist.web.api

import com.zorroa.archivist.HttpUtils
import com.zorroa.archivist.domain.DyHierarchy
import com.zorroa.archivist.domain.DyHierarchySpec
import com.zorroa.archivist.service.DyHierarchyService
import com.zorroa.archivist.service.FolderService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

/**
 * Created by chambers on 8/10/16.
 */
@RestController
class DyHierarchyController @Autowired constructor(
        val folderService : FolderService,
        val dyHierarchyService: DyHierarchyService
){

    @GetMapping(value = ["/api/v1/dyhi/_folder/{id}"])
    fun getByFolder(@PathVariable id: Int): DyHierarchy {
        val f = folderService.get(id)
        return dyHierarchyService.get(f)
    }

    @PostMapping(value = ["/api/v1/dyhi"])
    fun create(@RequestBody spec: DyHierarchySpec): DyHierarchy {
        return dyHierarchyService.create(spec)
    }

    @PostMapping(value = ["/api/v1/dyhi/{id}"])
    fun delete(@PathVariable id: Int): Map<String, Any> {
        val dh = dyHierarchyService.get(id)
        val result = dyHierarchyService.delete(dh)
        return HttpUtils.status("DyHierarchy", id, "delete", result)
    }

    @GetMapping(value = ["/api/v1/dyhi/{id}"])
    operator fun get(@PathVariable id: Int): DyHierarchy {
        return dyHierarchyService.get(id)
    }
}
